homeassistant/components/assist_pipeline/vad.py

```diff
@@ -239,3 +258,40 @@ def _process_chunk(self, chunk: bytes) -> bool:
             )
 
         return True
+
+
+def chunk_samples(
+    samples: bytes,
+    bytes_per_chunk: int,
+    sample_buffer: AudioBuffer,
+) -> Iterable[bytes]:
+    """Yield fixed-sized chunks from samples, keeping leftover bytes in a buffer."""
+    samples_offset = 0
+    num_chunks_in_samples = len(samples) // bytes_per_chunk
+    num_bytes_left = len(samples) % bytes_per_chunk
+
+    if len(sample_buffer) > 0:
+        # Add to partial chunk in buffer
+        bytes_to_copy = min(num_bytes_left, bytes_per_chunk - len(sample_buffer))
+        sample_buffer[len(sample_buffer) :] = samples[:bytes_to_copy]
+        sample_buffer.length += bytes_to_copy
```

Comment:
This is a bit weird, shouldn't this happen inside the `setitem` of `AudioBuffer` ?

Reply:
It's tricky because you could write over a portion of the existing buffer without change the "length" at all.
I think I'll add a helper method that make it clear what's happening here.